monai/transforms/croppad/array.py

```diff
@@ -1138,20 +1175,33 @@ def __init__(
         self.allow_smaller = allow_smaller
 
     def randomize(
-        self, label: torch.Tensor, indices: list[NdarrayOrTensor] | None = None, image: torch.Tensor | None = None
+        self,
+        label: torch.Tensor | None = None,
+        indices: list[NdarrayOrTensor] | None = None,
+        image: torch.Tensor | None = None,
     ) -> None:
-        indices_: Sequence[NdarrayOrTensor]
-        if indices is None:
-            if self.indices is not None:
-                indices_ = self.indices
-            else:
-                indices_ = map_classes_to_indices(label, self.num_classes, image, self.image_threshold)
+        _shape = None
+        if label is None:
+            raise ValueError("label must not be None.")
         else:
-            indices_ = indices
+            _shape = label.peek_pending_shape() if isinstance(label, MetaTensor) else label.shape[1:]
```

Comment:
try to simplify the logic here, if I understand correctly, seems the label can't be None here so I put it at the top.
https://github.com/Project-MONAI/MONAI/blob/ab6e83ec1ad919876b96fdc070a794d3d8f1ad72/monai/transforms/croppad/array.py#L1009-L1025

Reply:
I think when `indices_` is not None, `label`=None is allowed